README.md

```diff
@@ -197,34 +199,46 @@ The custom options that can be set on config param are detailed below.
-| _flushInterval_ | Defines the periodicity of buffer flushes in **miliseconds**. | `number` | `10000` | **NO** |
-| _flushSize_ | Defines the maximum buffer size before performing a flush. | `number` | `10` | **NO** |
+| _flushInterval_ | Defines the periodicity of buffer flushes in **miliseconds**. | `number` | `3000` | **NO** |
+| _flushSize_ | Defines the maximum buffer size before performing a flush. | `number` | `1000` | **NO** |
 | _namespace_ | Defines the global namespace. | `string` | `application` | **NO** |
 | _sampleRate_ | Defines the rate sampling. **Should be a number between [1, 100]**. | `number` | `100` | **NO** |
 | _tags_ | Defines the global tags. | `object` | `{}` | **NO** |
-| _transport_ | Defines the transport layer to be used to send metrics.<br><br> **Valid Transports:** `udp, http` | `string` | **none** | **YES** |
+| _transport_ | Defines the transport layer to be used to send metrics.<br><br> **Valid Transports:** `udp, api` | `string` | **none** | **YES** |
 | _host_ | Defines the host name to where the metrics should be sent. Can also be set inside _api_. | `string` | `127.0.0.1` | **NO** |
 | _port_ | Defines the port. Can also be set inside _api_. | `string` | `2013` | **NO** |
-| _secure_ | Enable or disable https protocol. Must be set inside _api_. | `boolean` | `true` | **NO** |
 | _token_ | Defines the token to be used.  Must be set inside _api_. | `string` | **none** | **NO** |
-| _timeout_ | Defines the timeout for the transport layers. Must be set inside _api_. | `number` | `2000` | **NO** |
+| _timeout_ | Defines the timeout for the transport layers in **miliseconds**. Must be set inside _api_. | `number` | `2000` | **NO** |
 
 ### Methods
 
 ```javascript
+// Non Aggregated Metrics
 - staful.counter('myCounter', 1, {agg: ['sum']});
 - staful.gauge('myGauge', 10, { tags: { host: 'localhost' } });
 - staful.timer('myCounter', 200, {namespace: 'sandbox'});
+- staful.put('myCustomMetric', 200, {timestamp: '1471519331'});
+
+// Aggregated Metrics
+- staful.aggregatedCounter('myCounter', 1, 'avg', 60, {agg: ['sum']});
+- staful.aggregatedGauge('myGauge', 10, 'avg', 60, { tags: { host: 'localhost' } });
+- staful.aggregatedTimer('myCounter', 200, 'avg', 60, {namespace: 'sandbox'});
+- staful.aggregatedPut('myCustomMetric', 200, 'avg', 60, {timestamp: '1471519331'});
 ```
-These methods receive a metric name and a metric value as arguments and send a counter/gauge/timer metric. If the options parameter is omitted, the default values are used.
+The methods for non aggregated metrics receive a metric name and a metric value as arguments and send a counter/gauge/timer/custom metric. 
+The methods for aggregated metrics receive a metric name, a metric value, an aggregation and an aggregation frequency (used previously to aggregate the metric) as arguments and send a counter/gauge/timer/custom metric.  
```

Comment:
I think we should make it a bit more clear on why we have these aggregated methods and the difference between them and the "main" methods - maybe separate it in a sub-section of the main methods section.